xarray/core/formatting.py

```diff
@@ -1023,7 +1023,7 @@ def diff_datatree_repr(a: DataTree, b: DataTree, compat):
 
 def _single_node_repr(node: DataTree) -> str:
     """Information about this node, not including its relationships to other nodes."""
-    node_info = f"DataTree('{node.name}')"
+    node_info = f"DataTree(name={node.name!r})"
```

Comment:
Adding the `name=` breaks a few tests, but I agree it is clearer

Reply:
Another option is to simply eliminate `name` when its value is `None`, so the repr for the empty tree looks like `DataTree(parent=None)` instead of `DataTree(name=None, parent=None)`.